inference-engine/tests/functional/plugin/shared/src/behavior/dynamic_batch.cpp

```diff
@@ -0,0 +1,178 @@
+#include "transformations/control_flow/unroll_tensor_iterator.hpp"
+#include "behavior/dynamic_batch.hpp"
+
+namespace BehaviorTestsDefinitions {
+
+    std::string DynamicBatchTest::getTestCaseName(const testing::TestParamInfo<dynamicBatchTestParams> &obj) {
+        std::string targetDevice;
+        InferenceEngine::Precision netPrecision;
+        std::vector<size_t> batchSizes;
+        bool runAsync;
+        std::map<std::string, std::string> config;
+        std::tie(targetDevice, netPrecision, batchSizes, runAsync, config) = obj.param;
+        std::ostringstream result;
+
+        result << "netPrecision=" << netPrecision.name() << "_";
+        result << "BS=" << CommonTestUtils::vec2str(batchSizes) << "_";
+        result << std::string(runAsync ? "Async" : "Sync") << "_";
+        result << "targetDevice=" << targetDevice;
+        return result.str();
+    }
+
+    size_t hiddenSize;
+
+
+    void DynamicBatchTest::SetUp() {
+        InferenceEngine::Precision netPrecision;
+        std::map<std::string, std::string> config;
+        std::tie(targetDevice, netPrecision, batch_sizes, run_async, config) = this->GetParam();
+        configuration.insert(config.begin(), config.end());
+        configuration[InferenceEngine::PluginConfigParams::KEY_DYN_BATCH_ENABLED] = InferenceEngine::PluginConfigParams::YES;
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+
+        max_batch_size = *std::max_element(batch_sizes.begin(), batch_sizes.end());
+
+        const int channel_number = 1;
+        const int inputSize = 32;
+
+        auto input = ngraph::builder::makeParams(ngPrc, { {channel_number, inputSize} , {channel_number, inputSize} , {channel_number, inputSize} });
+
+        auto relu_0 = ngraph::builder::makeActivation(input[0], ngPrc, ngraph::helpers::ActivationTypes::Relu);
+        auto relu_1 = ngraph::builder::makeActivation(input[1], ngPrc, ngraph::helpers::ActivationTypes::Relu);
+        auto relu_2 = ngraph::builder::makeActivation(input[2], ngPrc, ngraph::helpers::ActivationTypes::Relu);
+
+        auto concat_1 = ngraph::builder::makeConcat({relu_1, relu_0}, 1);
+        auto concat_2 = ngraph::builder::makeConcat({relu_2, relu_0}, 1);
+
+        auto eltw = ngraph::builder::makeEltwise(concat_1, concat_2, ngraph::helpers::EltwiseTypes::ADD);
+        auto sizes = eltw->get_shape();
+        auto reshape_pattern = std::make_shared<ngraph::op::Constant>(ngraph::element::i64, ngraph::Shape{2},
+                                                                      std::vector<size_t>({channel_number, inputSize * 2}));
+        auto final_reshape = std::make_shared<ngraph::op::v1::Reshape>(eltw, reshape_pattern, false);
+        function = std::make_shared<ngraph::Function>(final_reshape, input, "ActivationConcatsEltwise");
+    }
+
+    void DynamicBatchTest::Run() {
+        SKIP_IF_CURRENT_TEST_IS_DISABLED();
+        IE_SUPPRESS_DEPRECATED_START;
```

Comment:
Is it possible to remove this line? Why do use deprecated methods?

Reply:
Not really sure, copied it from other test to tell you the truth, will check if we can remove this.

Reply:
Done